src/runtime/Table.cpp

```diff
@@ -55,33 +55,37 @@ void Table::copy(ExecutionState& state, const Table* srcTable, uint32_t n, uint3
         }
         n--;
     }
+
+    return true;
 }
 
-void Table::fill(ExecutionState& state, uint32_t n, void* value, uint32_t index)
+bool Table::fill(ExecutionState& state, uint32_t n, void* value, uint32_t index)
 {
     if ((uint64_t)index + (uint64_t)n > (uint64_t)m_size) {
-        throwException(state);
+        return false;
     }
 
     while (n > 0) {
         m_elements[index] = value;
         n--;
         index++;
     }
+
+    return true;
 }
 
 void Table::throwException(ExecutionState& state) const
 {
     Trap::throwException(state, "out of bounds table access");
 }
 
-void Table::init(ExecutionState& state, Instance* instance, ElementSegment* source, uint32_t dstStart, uint32_t srcStart, uint32_t srcSize)
+bool Table::init(ExecutionState& state, Instance* instance, ElementSegment* source, uint32_t dstStart, uint32_t srcStart, uint32_t srcSize)
 {
     if (UNLIKELY((uint64_t)dstStart + (uint64_t)srcSize > (uint64_t)m_size)) {
-        throwException(state);
+        return false;
     }
     if (UNLIKELY(!source->element() || (srcStart + srcSize) > source->element()->functionIndex().size())) {
-        throwException(state);
+        return false;
     }
     if (UNLIKELY(m_type != Value::Type::FuncRef)) {
         Trap::throwException(state, "type mismatch");
```

Comment:
There is a `Trap::throwException` here. Please check that all throws are removed.